### PR TITLE
[FIX] sale_loyalty: check coupon partner

### PR DESCRIPTION
**Pull request by thle-odoo**
Steps to reproduce:
    - create a "gift card" with a specific customer;
    - use gift card code with any customer (Sales and/or eCommerce app).

Issue:
    The coupon is validated.

Cause:
    There is no partner verification when applying a coupon.

opw-3027251

### CODE REVIEW

File: addons/sale_loyalty/models/sale_order.py [hunk 1]
```diff
@@ -947,6 +947,8 @@ def _try_apply_code(self, code):
                 not coupon.program_id.reward_ids or\
                 not coupon.program_id.filtered_domain(self._get_program_domain()):
                 return {'error': _('This code is invalid (%s).', code), 'not_found': True}
+            elif coupon.partner_id.id and coupon.partner_id.id != self.partner_id.id:
```
Comment (agr-odoo) [inline addons/sale_loyalty/models/sale_order.py:950]:
```diff
- elif coupon.partner_id.id and coupon.partner_id.id != self.partner_id.id:
+ elif coupon.partner_id and coupon.partner_id != self.partner_id:

**Review by agr-odoo — COMMENTED** [review]
Looks ok. Beside the note made just for readability, we need also the new string in the `.pot` file

**Comment by agr-odoo** [comment]
robodoo r+

**Comment by dalaOdoo** [comment]
@thle-odoo & @agr-odoo this was made on purpose. We don't want to prevent a user of using a gift card code.
Here is the PR https://github.com/odoo/odoo/pull/99023
Can we revert this ?

**Comment by thle-odoo** [comment]
Revert PR: https://github.com/odoo/odoo/pull/103924

**Pull request closed without merging**